src/runtime/HalideBuffer.h

```diff
@@ -521,7 +556,9 @@ class Buffer {
     }
 
     /** Get the dimensionality of the buffer. */
+    // TODO: make constexpr, optimize for const case
```

Comment:
Why not handle this now. I think it's enough to just have an if (has_static_dimensions)

Reply:
Done.